client/src/pages/MyCart.jsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect, useContext } from "react"
+import { Link, useNavigate } from "react-router-dom";
+import { HiOutlineChevronLeft } from 'react-icons/hi'
+import { getTotalPrice, getTotalQuantity, AppContext } from "../lib";
+import { MyCartList, LoadingSpinner } from "../components";
+
+export function MyCart() {
+  const [cart, setCart] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const navigate = useNavigate();
+  const { user } = useContext(AppContext);
+
+  useEffect(() => {
+    !user && navigate('/signin');
+    const getCart = async () => {
+      try {
+        const token = localStorage.getItem('react-context-jwt')
```

Comment:
I'm seeing a lot of similar code that fetches or mutates data, either in the backend or in local storage. It would make the code cleaner and easier to maintain if you pulled all that out inside a separate lib file.